### PR TITLE
TestCafe topics

### LINKED ISSUE #435: Write about test framework
Topic about the test framework we use for UI tests

### PR DESCRIPTION
**Pull request by diana-lakatos**
Tutorial series based on @jnorbert's draft in #440. Includes all feedback from @pavelloz. 

Closes #435

### CODE REVIEW

**Review by jnorbert — APPROVED** [review]

File: marketplace_builder/views/pages/tutorials/qa/testcafe.liquid [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+  description: Introduction to TestCafe, a Node.JS tool for automating end-to-end web testing. 
+searchable: true
+converter: markdown
+---
+
+**[TestCafe](https://devexpress.github.io/testcafe)** is an open-source, Node.JS-based end-to-end testing framework for web apps. 
+
+## End-to-end testing
+
+**End-to-end testing** is a technique used to test whether the flow of an application right from start to finish is behaving as expected. The purpose of performing end-to-end testing is to identify system dependencies and to ensure that the data integrity is maintained between various system components and systems.
```
Comment (pavelloz) [inline marketplace_builder/views/pages/tutorials/qa/testcafe.liquid]: > End-to-end testing is a Software testing methodology to test an application flow from start to end. The purpose of End to end testing is to simulate the real user scenario and validate the system under test and its components for integration and data integrity.

I think this is better definition. It puts weight on "simulate real user scenario" which is the most important characteristic of those tests IMO

File: marketplace_builder/views/pages/tutorials/qa/testcafe.liquid [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+Together with unit and integration tests, end-to-end tests play an important role in  your quality assurance process, and can help you confirm that the application works as expected.
+
+## Why TestCafé?
```
Comment (pavelloz) [inline marketplace_builder/views/pages/tutorials/qa/testcafe.liquid]:
```suggestion
## Why TestCafe?
```

File: marketplace_builder/views/pages/tutorials/qa/testcafe.liquid (hunk 1 above)
Comment (diana-lakatos) [reply marketplace_builder/views/pages/tutorials/qa/testcafe.liquid]: Thank you!

File: marketplace_builder/views/pages/tutorials/qa/writing-testcafe-test.liquid [hunk 3]
```diff
@@ -0,0 +1,84 @@
+---
+slug: tutorials/qa/writing-testcafe-test
+metadata:
+  title: Writing a TestCafe Test
+  description: .
```
Comment (pavelloz) [inline marketplace_builder/views/pages/tutorials/qa/writing-testcafe-test.liquid]:
```suggestion
  description: This guide will help you write your first test using TestCafe. This tutorial uses an example from the platformOS documentation site. 
```

File: marketplace_builder/views/pages/tutorials/qa/testcafe.liquid (hunk 2 above)
Comment (diana-lakatos) [reply marketplace_builder/views/pages/tutorials/qa/testcafe.liquid]: Thanks!

**Review by pavelloz — APPROVED** [review]

**Comment by pavelloz** [comment]
LGTM :)

**Pull request merged**